kernels/typescript/default.nix

```diff
@@ -1,32 +1,57 @@
 {
   self,
   pkgs,
-  nodejs ? pkgs.nodejs-14_x,
+  npmlock2nix ? pkgs.npmlock2nix,
 }: let
-  inherit (pkgs) fetchurl lib stdenv writeScriptBin;
+  inherit (pkgs) lib stdenv writeScriptBin;
   inherit (lib) makeBinPath;
 
-  # This is honestly black magic and I forget where/how I discovered it.
-  tslab = let
-    inherit (lib) composeManyExtensions extends makeExtensible mapAttrs;
+  tslabSrc = fetchTarball {
+    url = "https://github.com/yunabe/tslab/archive/f056a88baf81a1fc6a01b0446a3febcf919ff8e4.tar.gz";
+    sha256 = "1q2wsdcgha6qivs238pysgmiabjhyflpd1bqbx0cgisgiz2nq3vs";
+  };
 
-    nodePackages = final:
-      import ./composition.nix {
-        inherit pkgs nodejs;
-        inherit (stdenv.hostPlatform) system;
-      };
+  tslab = pkgs.npmlock2nix.build {
+    src = tslabSrc;
+    node_modules_attrs.packageLockJson = ./package-lock.json;
+    buildInputs = [pkgs.makeWrapper];
+    postPatch = ''
+      # Change the source so it looks for the package.json file in the same
+      # directory level and not the parent. The package.json file will be
+      # stored in the same directory rather than the top level of the package.
+      substituteInPlace src/util.ts \
+        --replace "../package.json" "./package.json"
 
-    extensions = composeManyExtensions [
-      (import ./overrides.nix {inherit pkgs nodejs;})
-    ];
-  in
-    (makeExtensible (extends extensions nodePackages)).tslab;
+      # There is another file src/converter.ts that looks for the existence of
+      # the package.json file but appears that it will fail regardless and has
+      # a fallback so I do not think we need to patch this file.
+    '';
+    installPhase = ''
+      mkdir -p $out/bin
 
-  tslabSh = writeScriptBin "tslab" ''
-    #! ${stdenv.shell}
-    export PATH="${makeBinPath [tslab]}:$PATH"
-    ${tslab}/bin/tslab "$@"
-  '';
+      # Patch tslab binary so it looks for main in the correct location.
+      substitute bin/tslab $out/bin/tslab \
+        --replace "../dist/main.js" "../libexec/main.js"
+      chmod +x $out/bin/tslab
+
+      # Store distribution files in libexec as is standard for dependencies.
+      cp -r dist $out/libexec
+
+      # tslab needs the node modules at runtime but it is not available so we
+      # link it so the distribution files can find it.
+      ln -s ${tslab.node_modules}/node_modules $out/libexec/node_modules
+
+      # Wrap tslab so it knows where to find node modules and change the
+      # directory so it sees the directory.
+      wrapProgram $out/bin/tslab \
+        --set NODE_PATH $out/libexec/node_modules \
```

Comment:
Is `NODE_PATH` stiill required? It might not be now that we do the `chdir` thing